### PR TITLE
add docker image tests + run them in CI

### PR DESCRIPTION
**Pull request by aayars**
These are the old tests from cp-docker-images, updated to run without docker-machine.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,9 +14,10 @@
 # limitations under the License.
 
 # Builds a docker image for Confluent's example applications for the Kafka Streams API
-ARG DOCKER_REGISTRY
+ARG DOCKER_UPSTREAM_REGISTRY
```
Comment (ewencp) [inline Dockerfile]: should these have default values?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,5 +1,13 @@
 #!/usr/bin/env groovy
 
 docker_oraclejdk8 {
+    dockerRepos = ['confluentinc/kafka-streams-examples']
+    dockerRegistry = '368821881613.dkr.ecr.us-west-2.amazonaws.com/'
```
Comment (ewencp) [inline Jenkinsfile]: this should probably have some other dns name that we can redirect as we wish

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+git+https://github.com/confluentinc/confluent-docker-test-utils
```
Comment (ewencp) [inline requirements.txt]: this seems not ideal as it's not actually pinned to a version

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+from pip.req import parse_requirements
+import setuptools
+
+# Filters out relative/local requirements (i.e. ../lib/utils)
+remote_requirements = '\n'.join(str(r.req) for r in parse_requirements("requirements.txt", session='dummy') if r.req)
+
+setuptools.setup(
+    name='kafka-streams-examples-tests',
+    version='0.0.1',
```
Comment (ewencp) [inline setup.py]: can we make this just match the version in the pom? in fact, can we just get the version from the pom?

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    author="Confluent, Inc.",
+    author_email="partner-support@confluent.io",
```
Comment (ewencp) [inline setup.py]: this seems like it's not the right email address for this repo, shouldn't this be to either the streams team or some more general address? i don't think there is anything partner related in this repo

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+    description='Docker image tests',
+
+    url="https://github.com/confluentinc/kafka-connect-images",
```
Comment (ewencp) [inline setup.py]: wrong url for this repo?

File: test/fixtures/standalone-config.yml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 2181
+      ZOOKEEPER_TICK_TIME: 2000
+
+  kafka:
+    image: confluentinc/cp-kafka:latest
+    labels:
+    - io.confluent.docker.testing=true
+    environment:
+      KAFKA_BROKER_ID: 1
```
Comment (ewencp) [inline test/fixtures/standalone-config.yml]: not really related to this pr, but do we know if there is a reason we aren't using automatic broker ids?

File: test/test_kafka_streams_examples.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        # Verify access to the containerized application from inside its own container
+        self.is_kafka_music_app_healthy_for_service("kafka-streams-examples-bridge", port=17070)
+
+        # Verify outside access to the containerized application over the host network from a new container
+        list_running_app_instances_url_host = "http://{host}:{port}/kafka-music/instances".format(host="localhost", port=27070)
+        host_network_logs = utils.run_docker_command(
+            image="confluentinc/cp-kafka-streams-examples",
+            command=KAFKA_MUSIC_APP_HEALTH_CHECK.format(url=list_running_app_instances_url_host),
+            host_config={'NetworkMode': 'host'})
+        self.assertTrue("PASS" in host_network_logs)
```
Comment (ewencp) [inline test/test_kafka_streams_examples.py]: we seem to be mixing regular `assert`s and `self.assertTrue`

File: Dockerfile (hunk 1 above)
Comment (aayars) [reply Dockerfile]: If they're not given, the Docker default is to just use the latest from docker.io. We could provide defaults, but the implicit defaults are also okay.

File: Jenkinsfile (hunk 2 above)
Comment (aayars) [reply Jenkinsfile]: Unfortunately, this causes errors due to a principal name mismatch in the certificate :(

File: setup.py (hunk 4 above)
Comment (aayars) [reply setup.py]: The POM version is inherited from the parent -- we could just grab the parent version, but it's still pretty hacky.

I think we can just hard-code it to match the version for now.

File: Dockerfile (hunk 1 above)
Comment (ewencp) [reply Dockerfile]: huh, i wouldn't have thought including the `:` but not a tag following it would just use latest.

File: Jenkinsfile (hunk 2 above)
Comment (ewencp) [reply Jenkinsfile]: wow, how terrible and ugly. seems they've had requests for this for a couple of years now and still haven't implemented it.

**Comment by miguno** [comment]
I have nothing to add to this PR review. :-)

File: Dockerfile (hunk 1 above)
Comment (aayars) [reply Dockerfile]: You're right about tag, added a default.

File: setup.py (hunk 5 above)
Comment (aayars) [reply setup.py]: `pint` should be able to bump the POM and setup.py in one go.

**Pull request merged**